src/middleware/json-api/_deserialize.js

```diff
@@ -54,8 +54,8 @@ function resource (item, included, useCache = false) {
     var attrConfig = model.attributes[attr]
 
     if (_isUndefined(attrConfig) && attr !== 'id') {
-      attr = attr.replace(/-([a-z])/g, function (g) { return g[1].toUpperCase() })
-      attrConfig = model.attributes[attr]
+      const camelCaseAttr = attr.replace(/-([a-z])/g, function (g) { return g[1].toUpperCase() })
```

Comment:
I suspect that this breaks the deserializing of snake-case attributes unfortunately because `attr` was used in the same function below this line again to assign the value to the model.